doc/source/fq_nmod_sparse_mat.rst

```diff
@@ -0,0 +1,295 @@
+
+    Computes the decomposition PMQ = LU for a given sparse matrix ``M``, where 
+    ``P`` is a row permutation, ``Q`` is a column permutation, ``L``is a lower
+    triangular matrix, and ``U`` is an upper triangular matrix
+
+.. function:: void fq_nmod_sparse_mat_rref(fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Applies row reduction to put ``M`` in reduced row echelon form (in place)
+
+Solving
+--------------------------------------------------------------------------------
+
+.. function:: int fq_nmod_sparse_mat_solve_lu(fq_nmod_struct *x, const fq_nmod_sparse_mat_t M, const fq_nmod_struct *b, const fq_nmod_ctx_t ctx)
+
+    Given a matrix ``M`` and target vector ``b``, use LU decomposition to find
+    a vector ``x`` such that Mx = b, returns `1` if successful and `0` if not
+
+.. function:: int fq_nmod_sparse_mat_solve_rref(fq_nmod_struct *x, const fq_nmod_sparse_mat_t M, const fq_nmod_struct *b, const fq_nmod_ctx_t ctx)
+
+    Given a matrix ``M`` and target vector ``b``, use the reduced row-echelon
+    form to find a vector ``x`` such that Mx = b, returns `1` if successful and 
+    `0` if not
+
+.. function:: int fq_nmod_sparse_mat_solve_lanczos(fq_nmod_struct *x, const fq_nmod_sparse_mat_t M, const fq_nmod_struct *b, flint_rand_t state, const fq_nmod_ctx_t ctx)
+
+    Given a matrix ``M`` and target vector ``b``, use the Lanczos algorithm to
+    find a vector ``x`` such that Mx = b, returns `1` if successful and `0` if not
+
+.. function:: int fq_nmod_sparse_mat_solve_wiedemann(fq_nmod_struct *x, const fq_nmod_sparse_mat_t M, const fq_nmod_struct *b, const fq_nmod_ctx_t ctx)
+
+    Given a matrix ``M`` and target vector ``b``, use the Wiedemann algorithm to
+    find a vector ``x`` such that Mx = b, returns `1` if successful and `0` if not
+
+.. function:: int fq_nmod_sparse_mat_solve_block_lanczos(fq_nmod_struct *x, const fq_nmod_sparse_mat_t M, const fq_nmod_struct *b, slong block_size, flint_rand_t state, const fq_nmod_ctx_t ctx)
+
+    Given a matrix ``M`` and target vector ``b``, use Coppersmith's block Lanczos 
+    algorithm (with specified block size) to find a vector ``x`` such that Mx = b, 
+    returns `1` if successful and `0` if not
+
+.. function:: int fq_nmod_sparse_mat_solve_block_wiedemann(fq_nmod_struct *x, const fq_nmod_sparse_mat_t M, const fq_nmod_struct *b, slong block_size, flint_rand_t state, const fq_nmod_ctx_t ctx)
+
+    Given a matrix ``M`` and target vector ``b``, use Coppersmith's block Wiedemann
+    algorithm (with specified block size) to find a vector ``x`` such that Mx = b, 
+    returns `1` if successful and `0` if not
+
+Nullvector and nullspace computation
+--------------------------------------------------------------------------------
+
+.. function:: int fq_nmod_sparse_mat_nullvector_lanczos(fq_nmod_struct *x, const fq_nmod_sparse_mat_t M, flint_rand_t state, const fq_nmod_ctx_t ctx)
```

Comment:
The convention is for separate words to be separated by an underscore. To me, "nullvector" is not a single word. Also in the description.